Improve initial experience on OSX

miselin opened the pull request:
It should now be possible to run `make' on OSX without significant issues.

I need to also update the README to point out the existence of the new `Makefile` variables that are very useful depending on how the developer's OSX environment is configured.

miselin committed:
don't try and build debugdevice_full_duplex on osx

miselin committed:
pull in /usr/local for includes and libraries when building fx2 module

miselin committed:
update fx2load to use correct product id by default

mithro commented:
Hi @miselin,

Can you split the lights change into a separate pull request?

Tim 'mithro' Ansell

miselin committed:
added osx notes to readme

miselin commented:
No problems, I've removed the lights changes and will submit a separate PR for those shortly.

I've also added notes to the `README` about OSX; these could be moved to a separate `README.osx` file if they clutter the main `README` too much.

mithro commented:
BTW, Commit messages should start with a capital and end with a full stop :)
